interface.go

```diff
@@ -184,6 +184,9 @@ type Connection interface {
 	SendMessage([]byte) error
 	// ReceiveMessage gets a message received in a datagram, as specified in RFC 9221.
 	ReceiveMessage() ([]byte, error)
+
+	// GetVersion returns the QUIC version number used for this connection
+	GetVersion() protocol.VersionNumber
```

Comment:
Why not expose this on the ConnectionState?

Reply:
You've thought about this API more than I have, but here are two reasons:
1. The `ConnectionState` is a essentially a wrapper around `tls.connectionState` and the Version there refers to the TLS version use by the connection.
2. We already have a `GetVersion` method on the `connection` struct.

Reply:
> 1. The `ConnectionState` is a essentially a wrapper around `tls.connectionState` and the Version there refers to the TLS version use by the connection.

Not only, we also expose if the peer supports the datagram extension. I think it would make sense to add the QUIC version here, similar to how TLS exposes the version on the `tls.ConnectionState`.

> 2\. We already have a `GetVersion` method on the `connection` struct.

True. This is probably not very well-reasoned, but exposing another method feels like a bigger API change than adding a field to the `ConnectionState` struct.

Would you mind updating this PR?

Reply:
Sounds good! will do